src/scripts/painter.js

```diff
@@ -0,0 +1,74 @@
+(function() {
+  let ownColour;
+  let originalColour;
+  let reblogColour;
+  let likedColour;
+
+  const paint = async function() {
+    const { timelineObject } = await fakeImport('/src/util/react-props.js');
+
+    [...document.querySelectorAll('[data-id]:not(.xkit_painter_painted)')]
+    .forEach(async postElement => {
+      postElement.classList.add('xkit_painter_painted');
+
+      const post_id = postElement.dataset.id;
+      const { canDelete, liked, rebloggedFromId } = await timelineObject(post_id);
+
+      let coloursToApply = [];
+      if (canDelete && ownColour) {
+        coloursToApply.push(ownColour);
+      }
+      if (liked && likedColour) {
+        coloursToApply.push(likedColour);
+      }
+      if (rebloggedFromId) {
+        if (reblogColour) {
+          coloursToApply.push(reblogColour);
+        }
+      } else {
+        if (likedColour) {
+          coloursToApply.push(originalColour);
+        }
+      }
+
+      if (!coloursToApply.length) {
+        return;
+      }
+
+      const step = 100/coloursToApply.length;
+      let borderImage = 'linear-gradient(to right';
+      for (let i = 0; i < coloursToApply.length; i++) {
+        borderImage += `, ${coloursToApply[i]} ${step*i}% ${step*(i+1)}%`;
+      }
```

Comment:
i think this would be nicer as a `.forEach()`

```suggestion
      coloursToApply.forEach((colour, i) => 
        borderImage += `, ${colour} ${step*i}% ${step*(i+1)}%`
      );
```